### PR TITLE
code_800F7260

### PR DESCRIPTION
**Pull request by Thar0**
- 1 non-equivalent but is relatively close, but not close enough to verify equivalence due to a lot of regalloc induced by a few reorderings.
 - 1 left in asm, too many nested loops for me currently.
 - The rest of the file is matching.

The name `gSoundBanks` is taken from sm64 audio, and isn't really related to the file `Audiobank`. Could this possibly cause any confusion?

### CODE REVIEW

File: src/code/code_800F7260.c
Comment (krimtonz) [inline src/code/code_800F7260.c:148]: Interesting this was the method of breaking out of the loop

File: src/code/code_800F7260.c
Comment (krimtonz) [inline src/code/code_800F7260.c:177]:
```suggestion
                if (SFX_BANK_MASK(entry->sfxId) == SFX_BANK_MASK(arg1->unk_28) && (&entry->pos->x == arg1->posX)) {
```
Normally I view this as a preference thing, but with the & for the pointer on entry it looks a bit cluttered

**Review by krimtonz — COMMENTED** [review]
Looks pretty good to me, just a couple of minor things

File: src/code/code_800F7260.c
Comment (Thar0) [reply src/code/code_800F7260.c:148]: sm64 audio has a bunch of these reimplemented breaks too, seems like something to watch out for in other audio files as well

File: src/code/code_800F7260.c
Comment (Thar0) [reply src/code/code_800F7260.c:177]: I see what you mean with the `&& &` business, brackets seem like a good call here.

**Review by krimtonz — APPROVED** [review]

File: src/code/code_800F7260.c
Comment (krimtonz) [reply src/code/code_800F7260.c:148]: Does it actually set i to 10? Or just breaks?

Comment (Thar0) [reply src/code/code_800F7260.c:148]: It does yeah, it sets it to 10 then increments the loop counter as if it were about to continue, only stopping since it doesn't satisfy the condition to continue the loop. Replacing it with a proper `break` ruins codegen all over the function.

Comment (krimtonz) [reply src/code/code_800F7260.c:148]: I'd like to know what was going through their heads when they decided that

File: src/code/code_800F7260.c
Comment (Dragorn421) [inline src/code/code_800F7260.c:73]:
```suggestion
SoundBankEntry* gSoundBanks[7] = {
    D_8016BAD0, D_8016BC80, D_8016BEC0, D_8016C2E0, D_8016C6A0, D_8016C820, D_8016C8B0,
};

static u8 sBankSizes[ARRAY_COUNT(gSoundBanks)] = {
    ARRAY_COUNT(D_8016BAD0), ARRAY_COUNT(D_8016BC80), ARRAY_COUNT(D_8016BEC0), ARRAY_COUNT(D_8016C2E0), ARRAY_COUNT(D_8016C6A0), ARRAY_COUNT(D_8016C820), ARRAY_COUNT(D_8016C8B0),
};
```
`static` since you used the `s` prefix I guess this isn't used outside the file?

File: src/code/code_800F7260.c
Comment (Dragorn421) [inline src/code/code_800F7260.c:105]: imo `bankId` should be called an index (eg `bankIdx`) (indexes into an array) rather than an id which sounds more like map/dictionary/hashtable

File: src/code/code_800F7260.c
Comment (Dragorn421) [inline src/code/code_800F7260.c:277]:
```suggestion
            } else {
                return;
            }
        }
```

File: src/code/code_800F7260.c
Comment (Dragorn421) [inline src/code/code_800F7260.c:349]: `bankIndex` -> `entryIndex`?

File: src/code/code_800F7260.c
Comment (Dragorn421) [inline src/code/code_800F7260.c:554]: is this temp_f0 temp really mandatory (and temp_v0?)

File: include/z64audio.h
Comment (Dragorn421) [inline include/z64audio.h:761]:
```suggestion
    /* 0x2C */ u8       prev;       // prev entry index
    /* 0x2D */ u8       next;       // next entry index
```

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
I'm not convinced about the usage of "bank", I usually saw it used as a synonym for segment (as in "segmented address")
I have no clue how audio works though
I guess it can change later when more audio is decompiled and documented

Any clue what `SoundBankEntry` is? Is it a command that is overwritten when eg actors play sounds or parameters for playing sounds, or some kind of middle ground between the two?

File: src/code/code_800F7260.c
Comment (Thar0) [reply src/code/code_800F7260.c:73]: Other code files such as `game.c` use the `s` prefix without making the variables `static` so I'm also reluctant to do that currently as we might be doing that at a later stage.

Comment (fig02) [reply src/code/code_800F7260.c:73]: yeah the `s` prefix doesnt mean `static` keyword, its two separate uses for the same word. can just treat it as local vs global

File: src/code/code_800F7260.c
Comment (Thar0) [reply src/code/code_800F7260.c]: Both were removable it turns out

**Comment by Thar0** [comment]
> Any clue what `SoundBankEntry` is? Is it a command that is overwritten when eg actors play sounds or parameters for playing sounds, or some kind of middle ground between the two?

I'm not entirely sure but the rough equivalent in sm64 audio is `SoundCharacteristics`, so I believe it's the parameters of a sound request added through calls to public functions such as `Audio_PlaySoundGeneral`, which is indirectly linked to the actor system through wrappers such as `Audio_PlayActorSound2` in `z_actor.c` but is not limited to actors in itself e.g. being called directly to queue the sound that plays when advancing text.

**Review by fig02 — APPROVED** [review]

**Review by krimtonz — APPROVED** [review]

File: src/code/code_800F7260.c
Comment (Thar0) [reply src/code/code_800F7260.c:106]: After thinking about it for a while I think I'll hold off on naming changes in this PR unless there are very strong disagreements with current names and leave it to others who will do more audio than I will to rename when they learn more about the system than is currently known through other files.

**Review by Roman971 — APPROVED** [review]
I have no idea what's going on in there for now, so I'll trust yall with this.

**Pull request merged**
